src/store/store.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * @copyright 2019 Gary Kim <gary@garykim.dev>
+ *
+ * @author 2019 Gary Kim <gary@garykim.dev>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Vue from 'vue';
+import Vuex from 'vuex';
+import Axios from '@nextcloud/axios';
+import { generateUrl } from 'nextcloud-server/dist/router';
+
+Vue.use(Vuex);
+
+export default new Vuex.Store({
+	state: {
+		enabled: true,
+	},
+	mutations: {
+		enabled(state, val) {
+			state.enabled = val;
+		},
+	},
+	actions: {
+		enabled(ctx, val) {
+			ctx.commit('enabled', val);
+			Axios.put(generateUrl('apps/recommendations/settings/enabled'), {
+				value: val.toString(),
+			});
+		},
```

Comment:
```suggestion
		/**
		 * Toggle the recommendations
		 * @param {Object} context the store context
		 * @param {boolean} enabled recommendations status
		 */
		enabled(context, enabled) {
			context.commit('enabled', enabled);
			axios.put(generateUrl('apps/recommendations/settings/enabled'), {
				value: enabled.toString(),
			});
		},
```


Let's be consistent and easy to understand, we're compiling anyway, so bytes is not an issue here :stuck_out_tongue_closed_eyes: